server/settings_test.go

```diff
@@ -192,6 +196,56 @@ func TestSettings(t *testing.T) {
 				assert.True(t, resg.Payload.Settings.SttEnabled)
 			})
 
+			//Verify Failed checks alerts count in alertmanager
+			t.Run("VerifyFailedChecksInAlertmanager", func(t *testing.T) {
+				defer restoreDefaults(t)
+				// Enabling STT
+				res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
+					Body: server.ChangeSettingsBody{
+						EnableStt: true,
+					},
+					Context: pmmapitests.Context,
+				})
+				require.NoError(t, err)
+				assert.True(t, res.Payload.Settings.TelemetryEnabled)
+
+				activeAlerts := true
+				silencedAlerts := false
+				alertsCount := 0
+
+				//120 sec ping for failed checks alerts to appear in alertmanager
+				for i := 0; i < 120; i++ {
+					res, err := amclient.Default.Alert.GetAlerts(&alert.GetAlertsParams{
+						Active:   &activeAlerts,
+						Silenced: &silencedAlerts,
+						Context:  pmmapitests.Context,
+					})
+					require.NoError(t, err)
+					if len(res.Payload) != 0 {
+						for _, v := range res.Payload {
+							//verify that there is description in response
+							if _, ok := v.Annotations["description"]; !ok {
+								assert.True(t, false, "Description not met in response")
+							}
+							//verify that there is summary in response
+							if _, ok := v.Annotations["summary"]; !ok {
+								assert.True(t, false, "Summary not met in response")
+							}
+							//verify summary is not empty
+							if v.Annotations["summary"] == "" {
+								assert.True(t, false, "Summary is empty")
+							}
+						}
+						alertsCount = len(res.Payload)
+						break
+					}
+					time.Sleep(1 * time.Second)
+				}
+				msg := fmt.Sprint("No alerts met")
```

Comment:
Any reason to extract it as separate variable?

Reply:
oh no, i was building a string and forgot to change it. will change